src/test/java/org/owasp/html/EncodingTest.java

```diff
@@ -305,4 +371,66 @@ void testBadlyDonePostProcessingWillnotAllowInsertingNonceAttributes()
     Encoding.encodeHtmlAttribOnto("a nonce=xyz ", attrib);
     assertEquals("a nonce&#61;xyz ", attrib.toString());
   }
+
+  @Test
+  public static final void testRiskyNormalizationSetContents() {
```

Comment:
This unit test does the complete scan of the BMP for characters with a compatibility decomposition that could have side-effects.